smithy-model/src/main/java/software/amazon/smithy/model/shapes/OperationShape.java

```diff
@@ -180,7 +180,7 @@ public Builder addError(String errorShapeId) {
          * @param errorShapeIds Error shape IDs to add.
          * @return Returns the builder.
          */
-        public Builder addErrors(List<ShapeId> errorShapeIds) {
+        public Builder addErrors(Collection<ShapeId> errorShapeIds) {
```

Comment:
I know we have a thing in OperationIndex, but I feel like we should still have a helper method in OperationShape to get all errors and then deprecate `getErrors` or do something else to discourage its direct use.

Reply:
I think the shape is fine though. It's part of the metamodel and it returns the errors directly bound to the shape itself. It would be like deprecating an accessor in an AST which isn't a great idea -- we need to know the errors bound directly to the operation and emit them when serializing the model, for example.

I can add a documentation pointer to the index.

Reply:
I added another method like what you were suggesting -- `addErrors(ServiceShape service)`. It will return common errors of the given service plus operation errors.